driver-core/src/main/com/mongodb/client/model/Windows.java

```diff
@@ -0,0 +1,403 @@
+     *
+     * @param lower A value based on which the lower bound of the window is calculated.
+     * @param upper A value based on which the upper bound of the window is calculated.
+     * @return A range window.
+     */
+    public static Window range(final Decimal128 lower, final Bound upper) {
+        notNull("lower", lower);
+        notNull("upper", upper);
+        return new SimpleWindow<>(SimpleWindow.TYPE_RANGE_BASED, lower, upper.value(), null);
+    }
+
+    /**
+     * Creates a dynamically-sized range window whose bounds are determined by a range of possible values around
+     * the BSON {@link BsonType#DATE_TIME Date} value of the {@link Aggregates#setWindowFields(Object, Bson, List) sortBy}
+     * field in the current document.
+     *
+     * @param lower A value based on which the lower bound of the window is calculated.
+     * @param upper A value based on which the upper bound of the window is calculated.
+     * @param unit A time unit in which {@code lower} and {@code upper} are specified.
+     * @return A range window.
+     */
+    public static Window timeRange(final long lower, final long upper, final MongoTimeUnit unit) {
+        notNull("unit", unit);
+        isTrueArgument("lower <= upper", lower <= upper);
+        return new SimpleWindow<>(SimpleWindow.TYPE_RANGE_BASED, lower, upper, unit);
+    }
+
+    /**
+     * Creates a dynamically-sized range window whose bounds are determined by a range of possible values around
+     * the BSON {@link BsonType#DATE_TIME Date} value of the {@link Aggregates#setWindowFields(Object, Bson, List) sortBy}
+     * field in the current document.
+     *
+     * @param lower A value based on which the lower bound of the window is calculated.
+     * @param upper A value based on which the upper bound of the window is calculated.
+     * @param unit A time unit in which {@code upper} is specified.
+     * @return A range window.
+     */
+    public static Window timeRange(final Bound lower, final long upper, final MongoTimeUnit unit) {
+        notNull("lower", lower);
+        notNull("unit", unit);
+        return new SimpleWindow<>(SimpleWindow.TYPE_RANGE_BASED, lower.value(), upper, unit);
+    }
+
+    /**
+     * Creates a dynamically-sized range window whose bounds are determined by a range of possible values around
+     * the BSON {@link BsonType#DATE_TIME Date} value of the {@link Aggregates#setWindowFields(Object, Bson, List) sortBy}
+     * field in the current document.
+     *
+     * @param lower A value based on which the lower bound of the window is calculated.
+     * @param upper A value based on which the upper bound of the window is calculated.
+     * @param unit A time unit in which {@code lower} is specified.
+     * @return A range window.
+     */
+    public static Window timeRange(final long lower, final Bound upper, final MongoTimeUnit unit) {
```

Comment:
Do you think it is worth rearranging the parameters to be `lower`, `unit`, `upper` for this method? I am asking because currently the code looks like this

```java
timeRange(Windows.Bound.CURRENT, 1, , MongoTimeUnit.MONTH); // fine
timeRange(-1, Windows.Bound.CURRENT, MongoTimeUnit.MONTH); // a bit weird
```

if we change the order, the code would look like

```java
timeRange(Windows.Bound.CURRENT, 1, , MongoTimeUnit.MONTH); // fine
timeRange(-1, MongoTimeUnit.MONTH, Windows.Bound.CURRENT); // also fine
```

Reply:
Yes, that makes a lot of sense.  Go for it.

Reply:
Done.